app/models/sitemap.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class Sitemap < ActiveRecord::Base
+  RECENT_SITEMAP_NAME = 'recent'
+  NEWS_SITEMAP_NAME = 'news'
+
+  class << self
+    def regenerate_sitemaps
+      names_used = [RECENT_SITEMAP_NAME, NEWS_SITEMAP_NAME]
+
+      names_used.each { |name| touch(name) }
+
+      count = indexable_topics.count
+      max_page_size = SiteSetting.sitemap_page_size
+      size = count / max_page_size
+      size += 1 if count % max_page_size > 0
```

Comment:
Might be more readable, might be not 😛 
```suggestion
      size = (count.to_f / max_page_size).ceil
```

Reply:
Ended up using `divmod` instead. `ceil` didn't increment the number when the mod is greater than 0.